src/__tests__/convertIndexedToRgb.test.ts

```diff
@@ -0,0 +1,164 @@
+import { expect, describe, it } from 'vitest';
+
+import { decode, encode } from '..';
+import type { IndexedColors, DecodedPng } from '..';
+import { convertIndexedToRgb } from '../convertIndexedToRgb';
+
+import { loadAndDecode } from './decode.test';
+
+describe('rgb', () => {
+  it('1 bit', () => {
+    const palette: IndexedColors = [
+      [0, 0, 1],
+      [0, 0, 2],
+    ];
+    const decodedImage: DecodedPng = {
+      width: 1,
+      height: 1,
+      data: new Uint8Array([27]),
+      depth: 1,
+      palette,
+      channels: 1,
+      text: {},
+    };
+
+    const view = convertIndexedToRgb(decodedImage);
+    expect(view).toStrictEqual(
+      Uint8Array.from([
+        0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 2, 0, 0, 2, 0, 0, 1, 0, 0, 2, 0, 0, 2,
+      ]),
+    );
```

Comment:
Ok, now I don't understand. Since it's a 1x1 image, why does the returned data contain more than one color?

Reply:
I see. The algorithm only checks data and modifies it. Therefore, the values of width and height are irrelevant in this case. I can add a data check, that checks if the decoded png is a valid one, like it does during encoding.

Reply:
It can only check that the data array is long enough (and it probably should). A Uint8Array cannot contain less than one value. I think the image's size should be used by the algorithm to return a correctly sized array.

Reply:
Yeah, indeed. Just checking data length is not enough.